### PR TITLE
Update Team Management & User controllers

### PR DESCRIPTION
**Pull request by ajspotts**
Part 2 of stack
Part 1 #14981
Part 3 #14983

### Description
Updates the Team Management route & controller as well as the user controller to allow the new DVC team and members to be returned.

### Acceptance Criteria
- [ ] Tests pass

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -121,6 +121,26 @@
     end
   end
 
+  describe "GET /users?role=non_dvcs" do
+    before { dvc_team_count.times { DvcTeam.create_for_dvc(create(:user)) } }
+
+    context "when there are no dvc teams" do
+      let!(:users) { create_list(:user, 6) }
+      let(:dvc_team_count) { 0 }
```
Comment (hschallhorn) [inline spec/controllers/users_controller_spec.rb:129]: What is this testing? Looks like we're not creating any dvc teams to test that this correctly filters out users with dvc teams

File: spec/controllers/team_management_controller_spec.rb [hunk 2]
```diff
@@ -108,6 +110,38 @@
...
+
+    context "for a user who does not exist" do
+      let(:dvc_id) { "fake ID" }
+      it "returns a 404 error" do
+        post(:create_dvc_team, params: params, format: :json)
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context "for a user who already has a JudgeTeam" do
```
Comment (hschallhorn) [inline spec/controllers/team_management_controller_spec.rb:126]:
```suggestion
    context "for a user who already has a DvcTeam" do
```

Comment (ajspotts) [reply spec/controllers/team_management_controller_spec.rb:126]: 🙈

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (ajspotts) [reply spec/controllers/users_controller_spec.rb:129]: Tbh I was basing this test off the one we were already conducting for "non_judges" for the filter_by_role method in users_controller. We've added "non_dvcs" which returns the list of choosable users in the fe modal.  But I see your point, at best the test is saying "give me the users who don't have a DVC team" without ever explicitly creating one. I sanity checked myself just now and the filtering is indeed working properly (only users without a DVC team are being returned in the modal)

Comment (ajspotts) [reply spec/controllers/users_controller_spec.rb:129]: test added!

**Pull request merged**